tests/test_pysv.py

```diff
@@ -97,5 +101,52 @@ def incr(self, amount):
     run_tester(tester, target, simulator)
 
 
+def test_monitor(target, simulator):
+    class DelayedDUT(m.Circuit):
+        io = m.IO(A=m.In(m.Bits[4]), B=m.In(m.Bits[4]), O=m.Out(m.Bits[4]))
+        io += m.ClockIO(has_enable=True)
+        io.O @= m.Register(m.Bits[4], has_enable=True)()(dut()(io.A, io.B),
+                                                         CE=io.CE)
+
+    # TODO: Using a subclass breaks pysv
+    # class Monitor(fault.PysvMonitor):
+    class Monitor:
+        @sv()
+        def __init__(self):
+            self.value = None
+
+        @sv()
+        def observe(self, A, B, O):
+            print(A, B, self.value)
+            if self.value is not None:
+                assert O == self.value, f"{O} != {self.value}"
+            # TODO: Referencing bitvector here doesn't work, so just manually
+            # mask for now
+            # self.value = BitVector[4](A) + BitVector[4](B)
```

Comment:
Also, in the pysv github readme, there's an example using numpy inside the class, I tried this with hwtypes BitVector and it didn't work, is there something I need to do to enable library support?

```
<STDERR>
terminate called after throwing an instance of 'pybind11::error_already_set'
  what():  NameError: name 'BitVector' is not defined

At:
  <string>(12): observe

</STDERR>
```

Reply:
That sounds like a bug. I will try to reproduce it locally.

Reply:
Okay got it figured out:
1. Unfortunately importing foreign class is not currently supported since it requires more complex import mechanism. I've added that as an issue and will work on that once I get more free time: https://github.com/Kuree/pysv/issues/11. One workaround is to use the module.
2. The imports are cleared before the codegen. I think I added it for other tests to avoid importing modules because I couldn't get it working with the test environment. It should be removed now that the test environment is working.

Here is what I did to make it working locally on my machine:
```diff
diff --git a/tests/test_pysv.py b/tests/test_pysv.py
index 94803b4..20d2e61 100644
--- a/tests/test_pysv.py
+++ b/tests/test_pysv.py
@@ -5,6 +5,7 @@ import fault
 import magma as m
 from hwtypes import BitVector
 from .common import pytest_sim_params
+import hwtypes
 
 
 def pytest_generate_tests(metafunc):
@@ -123,12 +124,10 @@ def test_monitor(target, simulator):
                 assert O == self.value, f"{O} != {self.value}"
             # TODO: Referencing bitvector here doesn't work, so just manually
             # mask for now
-            # self.value = BitVector[4](A) + BitVector[4](B)
-            self.value = (A + B) & ((1 << 4) - 1)
+            self.value = hwtypes.BitVector[4](A) + hwtypes.BitVector[4](B)
+            # self.value = (A + B) & ((1 << 4) - 1)
             print(f"next value {self.value}")
 
-    clear_imports(Monitor)
-
     def test(circuit, enable):
         tester = fault.SynchronousTester(circuit)
         monitor = tester.Var("monitor", Monitor)

```

Reply:
Ignore the deleted comment if you've already seen it. I spoke too soon and forgot that I actually resolve the import limitation some time ago.

All you need is to remove `clear_imports`. I think I used it when I was hacking around the test environment to make the library load properly. Now that the testing environment is working, `clear_imports` should be removed.

Here is what I did locally to pass the test:
```diff
diff --git a/tests/test_pysv.py b/tests/test_pysv.py
index 94803b4..8e6b7f8 100644
--- a/tests/test_pysv.py
+++ b/tests/test_pysv.py
@@ -123,12 +123,10 @@ def test_monitor(target, simulator):
                 assert O == self.value, f"{O} != {self.value}"
             # TODO: Referencing bitvector here doesn't work, so just manually
             # mask for now
-            # self.value = BitVector[4](A) + BitVector[4](B)
-            self.value = (A + B) & ((1 << 4) - 1)
+            self.value = BitVector[4](A) + BitVector[4](B)
+            # self.value = (A + B) & ((1 << 4) - 1)
             print(f"next value {self.value}")
 
-    clear_imports(Monitor)
-
     def test(circuit, enable):
         tester = fault.SynchronousTester(circuit)
         monitor = tester.Var("monitor", Monitor)

```